cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsStackRequestHelper.java

```diff
@@ -117,33 +118,34 @@ private String getRootDeviceName(AuthenticatedContext ac, CloudStack cloudStack)
         return image.getRootDeviceName();
     }
 
-    private Collection<Parameter> getStackParameters(AuthenticatedContext ac, DatabaseStack stack) {
+    @VisibleForTesting
+    Collection<Parameter> getStackParameters(AuthenticatedContext ac, DatabaseStack stack) {
         AwsNetworkView awsNetworkView = new AwsNetworkView(stack.getNetwork());
         AwsRdsInstanceView awsRdsInstanceView = new AwsRdsInstanceView(stack.getDatabaseServer());
         AwsRdsDbSubnetGroupView awsRdsDbSubnetGroupView = new AwsRdsDbSubnetGroupView(stack.getDatabaseServer());
         List<Parameter> parameters = new ArrayList<>(asList(
-                // FIXME allocated storage might be null
-                new Parameter().withParameterKey("AllocatedStorageParameter").withParameterValue(Objects.toString(awsRdsInstanceView.getAllocatedStorage())),
-                new Parameter().withParameterKey("BackupRetentionPeriodParameter")
-                    .withParameterValue(Objects.toString(awsRdsInstanceView.getBackupRetentionPeriod())),
                 new Parameter().withParameterKey("DBInstanceClassParameter").withParameterValue(awsRdsInstanceView.getDBInstanceClass()),
                 new Parameter().withParameterKey("DBInstanceIdentifierParameter").withParameterValue(awsRdsInstanceView.getDBInstanceIdentifier()),
                 new Parameter().withParameterKey("DBSubnetGroupNameParameter").withParameterValue(awsRdsDbSubnetGroupView.getDBSubnetGroupName()),
                 new Parameter().withParameterKey("DBSubnetGroupSubnetIdsParameter").withParameterValue(String.join(",", awsNetworkView.getSubnetList())),
                 new Parameter().withParameterKey("EngineParameter").withParameterValue(awsRdsInstanceView.getEngine()),
-                new Parameter().withParameterKey("EngineVersionParameter").withParameterValue(awsRdsInstanceView.getEngineVersion()),
                 new Parameter().withParameterKey("MasterUsernameParameter").withParameterValue(awsRdsInstanceView.getMasterUsername()),
                 new Parameter().withParameterKey("MasterUserPasswordParameter").withParameterValue(awsRdsInstanceView.getMasterUserPassword()),
                 new Parameter().withParameterKey("VPCSecurityGroupsParameter").withParameterValue(String.join(",", awsRdsInstanceView.getVPCSecurityGroups())),
-                new Parameter().withParameterKey("StackOwner").withParameterValue(String.valueOf(ac.getCloudContext().getUserName()))
+                new Parameter().withParameterKey("StackOwner").withParameterValue(ac.getCloudContext().getUserName())
         ));
 
-        // Add the port if it's been supplied
-        Integer port = stack.getDatabaseServer().getPort();
-        if (port != null) {
-            parameters.add(new Parameter().withParameterKey("PortParameter").withParameterValue(Objects.toString(port)));
-        }
+        addParameterIfNotNull(parameters, "AllocatedStorageParameter", awsRdsInstanceView.getAllocatedStorage());
```

Comment:
Just a question: do all the values made optional here have a default value in aws?

Reply:
They have default values in the CloudFormation template, actually. I think allocated storage is required by the RDS API, but the CF template will always supply something.